src/client/components/layout.js

```diff
@@ -0,0 +1,141 @@
+					<span className="sr-only">Toggle navigation</span>
+					<span className="icon-bar"></span>
+					<span className="icon-bar"></span>
+					<span className="icon-bar"></span>
+				</a>
+				<a href="/" className="navbar-brand logo">
+					{homepage ?
+						<img alt="BookBrainz icon" src="/images/BookBrainz_logo_icon.svg" title="BookBrainz"/>
+						:
+						<img alt="BookBrainz icon" src="/images/BookBrainz_logo_mini.svg" title="BookBrainz"/>
+					}
+				</a>
+			</div>
+		);
+
+		const navContent = (
+			<div id="bs-example-navbar-collapse-1" className="collapse navbar-collapse">
+				<ul className="nav navbar-nav navbar-right">
+					{user && user.id ? 
+						<div>
+							<li className="dropdown">
+								<a id="dNewEntities" href="#" data-toggle="dropdown" role="button" aria-expanded="false" className="dropdown-toggle">
+									<span className="fa fa-plus"></span>&nbsp;Create&nbsp;
+									<span className="caret"></span>
+								</a>
+								<ul role="menu" aria-labelledby="dNewEntities" className="dropdown-menu">
+									<li><a href="/publication/create">Create Publication</a></li>
+									<li><a href="/edition/create">Create Edition</a></li>
+									<li><a href="/work/create">Create Work</a></li>
+									<li className="divider"></li>
+									<li><a href="/creator/create">Create Creator</a></li>
+									<li><a href="/publisher/create">Create Publisher</a></li>
+								</ul>
+							</li>
+							<li className="dropdown">
+								<a id="dUserDropdown" href="#" data-toggle="dropdown" role="button" aria-expanded="false" className="dropdown-toggle">
+									<span className="fa fa-user"></span><span>&nbsp; {user.name}</span><span className="caret"></span>
+								</a>
+								<ul role="menu" aria-labelledby="dUserDropdown" className="dropdown-menu">
+									<li><a href={`/editor/${user.id}`}><span className="fa fa-info fa-fw"></span>&nbsp;Profile</a></li>
+									<li><a href="/logout"><span className="fa fa-sign-out fa-fw"></span>&nbsp;Sign Out</a></li>
+								</ul>
+							</li>
+						</div>
+						:
+						<li>
+							<a href="/auth"><span className="fa fa-sign-in"></span>&nbsp;Sign In / Register</a>
+						</li>
+					}
+				</ul>
+				{!(homepage || hideSearch) && 
+				<form role="search" action="/search" className="navbar-form navbar-right">
+					<div className="form-group">
+						<div className="input-group">
+						<input type="text" placeholder="Search for..." name="q" className="form-control"/><span className="input-group-btn">
+							<button type="submit" className="btn btn-success"><span className="fa fa-search"></span></button></span>
+						</div>
+					</div>
+				</form>
+			}
+		</div>
+
+	);
+
+		const children = homepage ? 
+		React.Children.map( this.props.children, child => React.cloneElement(child, {...this.props}))
+		:
+		(<div id="content" className="container">
+			{ React.Children.map( this.props.children, child => React.cloneElement(child, {...this.props}))}
```

Comment:
This is being used for prop injection. I intend on refactoring it to make it clearer and more efficient.